packages/hurl/src/http/response_debug.rs

```diff
@@ -34,7 +35,22 @@ impl Response {
             }
         }
         match self.text() {
-            Ok(text) => debug::log_text(&text, debug, logger),
+            Ok(text) => match serde_json::from_str::<Value>(&text) {
```

Comment:
I would apply the JSON pretty print only if the response has the `application/json` content-type.

Reply:
Also if the response has the `application/json` content-type with an invalid JSON body, I would expect a runtime error if `--pretty` is used.